compiler/rustc_mir_build/src/build/matches/mod.rs

```diff
@@ -1536,10 +1536,13 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         start_block: BasicBlock,
         candidates: &'b mut [&'c mut Candidate<'pat, 'tcx>],
     ) -> BlockAnd<&'b mut [&'c mut Candidate<'pat, 'tcx>]> {
-        // We can't expand or-patterns freely. The rule is: if the candidate has an
-        // or-pattern as its only remaining match pair, we can expand it freely. If it has
-        // other match pairs, we can expand it but we can't process more candidates after
-        // it.
+        // We can't expand or-patterns freely. The rule is:
+        // - If a candidate doesn't start with an or-pattern, we include it in
+        //   the expansion list but don't expand it.
```

Comment:
"but don't expand it" feels confusing to me given that there's nothing that we could expand. Maybe "we add it to the expansion list and it will be processed alongside the expanded or-patterns"

Reply:
Yeah, it's tricky to express what's happening here without being misleading/confusing. Hopefully the new wording is better.